x-pack/auditbeat/magefile.go

```diff
@@ -82,8 +83,13 @@ func TestPackages() error {
 }
 
 // Update is an alias for running fields, dashboards, config.
-func Update() {
+func Update() error {
+	err := sh.Run("make", "-C", "../../auditbeat", "update")
```

Comment:
@andrewkroh I had to introduce this hack as for the generation of the x-pack Beats fields.yml they depend on the build/fields/* files in the OSS Beats.

This also made me think if Auditbeat does the right thing here. As it depends on Metricbeat it should also include the `beat.yml` data and in this case depending on metricbeat update? Does priority matter here? I would assume it's ok that the metricbeat and auditbeat beat.yml have the same priority as they should not conflict with each other. To be save, would be nice to have a way to set different priorities here.

All the above becomes obsolete as soon as we generate the `fields.yml` by `beat export fields` as then each beat can only generate it's own files and it then depends on the internal dependency / priority how the fields.yml is exported.

Reply:
Why does x-pack/auditbeat depend on anything in OSS auditbeat/build/fields? That should not be the case. When building fields.yml files in x-pack/auditbeat the should use the checked in source to derive things. This is how fields and dashboards work. (It's been a while since I last looked at this PR and it's Friday so I could be missing something. Maybe should should discuss this part next week.)

Reply:
I need to check again, but I remember some fields / fields from OSS were missing otherwise. Will get back to you on this one as soon as I got the time.

Reply:
The problem is around here: https://github.com/elastic/beats/pull/9507/files#diff-408f5df10fc1cfd92fc66134de4d8edbR77 The generation of the fields.yml still works, but it can't create the new fields.go files. The problem is that currently the `build/fields/*.yml` files are only built for OSS. This becomes now more complex, as X-Pack needs to also build these files to then derive the modules and potentially specific beats files for the new local fields.go files. Need to think through on how we deal with it best. The x-pack beat does not need knowledge about the beat fields of the non x-pack one and can also skip libbeat.